### PR TITLE
LPS-162321 Poshi Automation for Bug | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-162321

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 1]
```diff
@@ -517,6 +517,91 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-158221. The Asset Publisher widget should not lose the layout article reference after edit associated content page."
+	@priority = "4"
+	test EditAssociatedContentPageShouldNotDeleteLayoutReference {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:522]: test RepublishAssociatedContentPage

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 2]
```diff
@@ -517,6 +517,91 @@ definition {
...
+		task ("Given a page creator has a displayed web content on associated Asset Publisher widget") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:523]: Given a page creator has a web content with an associated content page

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 3]
```diff
@@ -517,6 +517,91 @@ definition {
...
+						pageName = "Test Page Name ${i}",
+						siteName = "Test Site Name");
+
+					PageEditor.addWidget(portletName = "Asset Publisher");
+
+					PageEditor.publish();
+				}
+			}
+
+			task ("Select Content Editing") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:545]: ContentPagesNavigator.openEditContentPage(
						pageName = "Test Page Name 2",
						siteName = "Test Site Name");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 4]
```diff
@@ -517,6 +517,91 @@ definition {
...
+
+				WebContent.editDisplayPage(
+					pageName = "Test Page Name 2",
+					pageType = "Pages",
+					webContentTitle = "Web Content Title");
+
+				PortletEntry.publish();
+			}
+
+			task ("Open the associated content page at edit mode and publish it") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:579]: When the page creator republish the associated content page

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase [hunk 5]
```diff
@@ -517,6 +517,91 @@ definition {
...
+			AssetPublisherPortlet.clickAssetTitle(assetTitle = "Web Content Title");
+		}
+
+		task ("Then the page creator should be redirected to associated display page") {
+			AssetPublisherPortlet.viewAssetPG(
+				assetContent = "Web Content Content",
+				assetTitle = "Web Content Title");
+
+			ControlMenu.viewHeaderTitle(headerTitle = "Test Page Name 2");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherUseCase.testcase:602]: Then page creator should be redirected to the associated content page after clicking the web content title on Asset Publisher

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating according to comments? Thanks

Hi @Tim-Cao, PR updated. Could you please have a look at? Thanks!

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**